app/controllers/userController.js

```diff
@@ -0,0 +1,190 @@
+const User = require("../models/user.model");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+const Joi = require("joi");
+
+const signup = async (req, res) => {
+  const { firstName, lastName, phoneNumber, email, password } = req.body;
+
+  // Define a schema for data validation
+  const schema = Joi.object({
+    firstName: Joi.string(),
+    lastName: Joi.string(),
+    phoneNumber: Joi.string().regex(/^\d+$/), // Ensure only digits are allowed
+    email: Joi.string().email(),
+    password: Joi.string().min(6),
+  });
+
+  try {
+    // Validate user input against the schema
+    const { error } = schema.validate(req.body);
+
+    if (error) {
+      return res.status(400).json({ error: error?.details[0]?.message });
+    }
+
+    // Check if the user already exists
+    const existingUser = await User.findOne({ email });
+
+    if (existingUser) {
+      return res.status(400).json({ error: "User already exists" });
+    }
+
+    // Create a new user
+    const newUser = new User({
+      firstName,
+      lastName,
+      phoneNumber,
+      email,
+      password,
+    });
+
+    // Save the user to the database
+    await newUser.save();
+
+    res.status(200).json({ message: "User registered successfully" });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: "An error occurred during signup" });
+  }
```

Comment:
For error use one error handler on index.js file. 
from here error should be send using next(err) and your error middleware will take care of sending status and message.

Also, make sure to pass the statusfor different error which are sending here.

Reply:
do this for all the API below